tests/matchers/test_iou.py

```diff
@@ -15,146 +15,146 @@
-    # Create match tuples
-    matches = list(zip(gtcells, rescells))
-    # Check for direct match
-    assert (1, 3) in matches
-    # Check that only one of the merge matches is present
-    assert ((2, 4) in matches) != ((2, 5) in matches)
+    @pytest.mark.parametrize(
+        "data_fxn",
+        [ex_segs.oversegmentation_2d, ex_segs.oversegmentation_3d],
+        ids=["2D", "3D"],
+    )
+    def test_split(self, data_fxn):
+        gt, pred = data_fxn()
+
+        # Low threshold, both match
+        expected_matches = [(1, 2), (1, 3)]
+        gtcells, rescells = _match_nodes(gt, pred, threshold=0.3)
+        assert Counter(expected_matches) == Counter(list(zip(gtcells, rescells)))
+
+        # High threshold, no match
+        expected_matches = []
+        gtcells, rescells = _match_nodes(gt, pred, threshold=0.7)
+        assert Counter(expected_matches) == Counter(list(zip(gtcells, rescells)))
+
+        # Low threshold, one to one, only one matches
+        gtcells, rescells = _match_nodes(gt, pred, threshold=0.3, one_to_one=True)
+        computed_matches = list(zip(gtcells, rescells))
+        assert ((1, 2) in computed_matches) != ((1, 3) in computed_matches)
+
+    @pytest.mark.parametrize(
+        "data_fxn",
+        [ex_segs.undersegmentation_2d, ex_segs.undersegmentation_3d],
+        ids=["2D", "3D"],
+    )
+    def test_merge(self, data_fxn):
+        gt, pred = data_fxn()
+
+        # Low threshold, both match
+        expected_matches = [(1, 3), (2, 3)]
+        gtcells, rescells = _match_nodes(gt, pred, threshold=0.3)
+        assert Counter(expected_matches) == Counter(list(zip(gtcells, rescells)))
+
+        # High threshold, no match
+        expected_matches = []
+        gtcells, rescells = _match_nodes(gt, pred, threshold=0.7)
+        assert Counter(expected_matches) == Counter(list(zip(gtcells, rescells)))
+
+        # Low threshold, one to one, only one matches
+        gtcells, rescells = _match_nodes(gt, pred, threshold=0.3, one_to_one=True)
+        computed_matches = list(zip(gtcells, rescells))
+        assert ((1, 3) in computed_matches) != ((2, 3) in computed_matches)
+
+    @pytest.mark.parametrize(
+        "data_fxn", [ex_segs.multicell_2d, ex_segs.multicell_3d], ids=["2D", "3D"]
+    )
+    def test_multiple_objects(self, data_fxn):
+        gt, pred = data_fxn()
+        expected_matches = [(1, 3)]
+        gtcells, rescells = _match_nodes(gt, pred)
+        assert Counter(expected_matches) == Counter(list(zip(gtcells, rescells)))
+
+    @pytest.mark.parametrize(
+        "data_fxn", [ex_segs.no_overlap_2d, ex_segs.no_overlap_3d], ids=["2D", "3D"]
+    )
+    def test_no_overlap(self, data_fxn):
+        gt, pred = data_fxn()
+        expected_matches = []
+        gtcells, rescells = _match_nodes(gt, pred)
+        assert Counter(expected_matches) == Counter(list(zip(gtcells, rescells)))
+
+    def test_input_error(self):
+        im = np.zeros((10, 10))
+        with pytest.raises(ValueError):
```

Comment:
Maybe we can use the `match` argument for `raises` here to ensure the error message corresponds to the specific error we're expecting i.e. 

```suggestion
        with pytest.raises(ValueError, match="Threshold of 0 is not valid unless one_to_one is True"):
```